utils/amt-integration-tests/tests/data_helpers.py

```diff
@@ -0,0 +1,163 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+import os
+
+import pandas as pd
+from sqlalchemy import engine, text
+
+from .table_helpers import read_keyvalue_pairs_as_dataframe
+
+
+DESCRIPTOR_SQL = """
+INSERT INTO edfi.Descriptor (CodeValue, ShortDescription, Description, Namespace)
+    VALUES ('<cv>', '<sd>', '<d>', '<n>');
+DECLARE @id int = @@identity;
+INSERT INTO <schema>.<tbl> (<tbl>Id) VALUES (@id);
+"""
+
+SCHEMA_LMSX = "lmsx"
+SCHEMA_EDFI = "edfi"
+
+
+def _get_value(row: pd.Series, column: str) -> str:
+    # Escape any apostrophes so that this is safe to run
+    return row[column].replace("'", "''")
+
+
+def _get_edfi_options(engine: engine.base.Engine) -> dict:
+    return {"con": engine, "schema": SCHEMA_EDFI, **APPEND_OPTIONS}
+
+
+def _prepare_descriptor_sql(row: pd.Series, schema: str, table: str) -> str:
+    return (
+        DESCRIPTOR_SQL.replace("<cv>", _get_value(row, "CodeValue"))
+        .replace("<sd>", _get_value(row, "ShortDescription"))
+        .replace("<d>", _get_value(row, "Description"))
+        .replace("<n>", _get_value(row, "Namespace"))
+        .replace("<tbl>", table)
+        .replace("<schema>", schema)
+    )
+
+
+def _read_and_load_descriptors(
+    engine: engine.base.Engine, descriptor_type: str
+) -> None:
+    descriptor = f"{descriptor_type}Descriptor"
+
+    file_path = os.path.join(
+        "..", "..", "extension", "Descriptors", f"{descriptor}.xml"
+    )
+    df = pd.read_xml(file_path)
+
+    with engine.connect() as connection:
+        for _, row in df.iterrows():
+            sql = _prepare_descriptor_sql(row, SCHEMA_LMSX, descriptor)
+            connection.execute(text(sql))
+
+
+def load_lms_descriptors(engine: engine.base.Engine) -> None:
+    _read_and_load_descriptors(engine, "AssignmentCategory")
+    _read_and_load_descriptors(engine, "LMSSourceSystem")
+    _read_and_load_descriptors(engine, "SubmissionStatus")
+    _read_and_load_descriptors(engine, "SubmissionType")
+
+
+APPEND_OPTIONS = {"if_exists": "append", "index": False}
+
+
+def load_school(engine: engine.base.Engine, id: str) -> None:
+    ed_org = pd.DataFrame([{"EducationOrganizationId": id, "NameOfInstitution": id}])
+    ed_org.to_sql("EducationOrganization", **_get_edfi_options(engine))
+
+    school = pd.DataFrame([{"schoolid": id}])
+    school.to_sql("School", **_get_edfi_options(engine))
```

Comment:
Why are you using a dataframe instead of an engine.execute("sql script") or something like that?

Reply:
Because this is easier than the alternative of writing a SQL statement by hand. Just look at how little code was needed to get this working ;-). Oh, and it will also work on either PostgreSQL or SQL Server, whereas if it was hand written it might only work on SQL Server.